ProtocolHandler.pm

```diff
@@ -269,10 +269,20 @@ sub getMetadataFor {
 
 	$meta->{title} = Plugins::Qobuz::API::Common->addVersionToTitle($meta);
 	
-	if ($meta->{work}) {
-		$meta->{title} = $meta->{work} . ': ' . $meta->{title};
+	my $genreList = $prefs->get('classicalGenres');
+	if ( $prefs->get('useClassicalEnhancements') 
+		&& ( grep(/Classique/,@{$meta->{genres_list}}) || grep(/(^\s*$meta->{genre}\s*$|^\s*\*$)/,(split ',', $genreList)) )
```

Comment:
You should write a small helper function to do this grepping. It's pretty complicated, needs another check, and you use it more than once.

Once it's in a single place you could also consider optimizing it: pre-split the scalar genre list into a list, or even a hash for quicker checks.